### ISSUE TITLE
Issue with `at-rule-semicolon-space-before:never` when no semicolon in use

### ISSUE THREAD

**Issue by u01jmg3** [stylelint/stylelint#6363]
### What steps are needed to reproduce the bug?

1. 

```sh
npm install postcss postcss-scss stylelint --save-dev
```

2.

```sh
stylelint test.scss
```

```scss
// test.scss
.padding-bottom-15 { padding-bottom: 15px }

.element { @extend .padding-bottom-15 }
```

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
    customSyntax: 'postcss-scss',
    rules: {
        'at-rule-semicolon-space-before': 'never',
    },
};
```

### How did you run Stylelint?

CLI with `stylelint test.scss`

### Which version of Stylelint are you using?

`14.12.1`

### What did you expect to happen?

There should be no output from Stylelint as no semicolons are in use.

### What actually happened?

The following problem was reported:

```
test.scss
 3:39  ✖  Unexpected whitespace before ";"  at-rule-semicolon-space-before
```

I have found that I can stop the error by adding a semicolon but this shouldn't be necessary.

```diff
// test.scss
.padding-bottom-15 { padding-bottom: 15px }

-.element { @extend .padding-bottom-15 }
+.element { @extend .padding-bottom-15; }
```

### Does the bug relate to non-standard syntax?

Yes, it _could_ be related to the use of `@extend` (SCSS)

### Proposal to fix the bug

Unfamiliar with the code but happy to take a look if nobody else can spot what the issue might be

**Comment by u01jmg3** [comment]
Maybe @CAYdenberg knows as the original author?

**Comment by jeddy3** [comment]
@u01jmg3 Thanks for the clearly written report and for using the template.

We've [frozen the stylistic rules](https://stylelint.io/user-guide/rules/list#stylistic-issues). We now recommend using a pretty printer alongside Stylelint, such as Prettier. Alternatively, if you wish to continue using Stylelint for stylistic stuff (rather than for only [avoiding errors](https://stylelint.io/user-guide/rules/list#avoid-errors) and [enforcing non-stylistic conventions](https://stylelint.io/user-guide/rules/list#enforce-conventions)) you can [port the rule to a plugin](https://stylelint.io/developer-guide/plugins) and fix the bug there.

It may also be an issue with the [PostCSS SCSS syntax](https://github.com/postcss/postcss-scss). If it is, then may be able to address the issue upstream.

**Comment by CAYdenberg** [comment]
I don't remember writing this code, but it seems pretty clear this is a bug, no? (even if this rule isn't something Stylelint _should_ still have, this is clearly wrong). I can take a look later today.
